spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala

```diff
@@ -110,13 +124,14 @@ class BigQueryStrategy(expressionConverter: SparkExpressionConverter, expression
    *         unsupported node type.
    */
   def generateQueryFromPlan(plan: LogicalPlan): Option[BigQuerySQLQuery] = {
-    plan match {
+    val updatedPlan = cleanUpLogicalPlan(plan)
```

Comment:
Do we ALWAYS need to skip this node? @VinayLondhe14

Reply:
We actually need to clean up the plan (and skip this node) at the beginning of the traversal process.

@suryasoma can you please move this method call in `generateSparkPlanFromLogicalPlan`?